src/main/java/com/example/config/JdbcTemplateConfig.java

```diff
@@ -0,0 +1,23 @@
+package com.example.config;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.jdbc.core.JdbcTemplate;
+
+import javax.sql.DataSource;
+
+@Configuration
+@ComponentScan(basePackages = "com.example")
```

Comment:
no need to duplicate the component scan each and every time. You already specified it in the DatasourceConfig.java